lib/src/state_machine.dart

```diff
@@ -117,3 +135,53 @@ class State<StateIdentifier> {
   /// and [actions] associated with that state.
   State(this.identifier, {required this.events, required this.actions});
 }
+
+/// A state diagram generator for FSM.
+///
+/// Outputs to vcd format at [outputPath].
+class MermaidStateDiagram {
+  /// The diagram to be return as String
+  late String diagram;
+
+  /// The output filepath of the generated state diagram.
+  final String outputPath;
+
+  /// The file to write dumped output waveform to.
+  final File _outputFile;
+
+  /// Generate a [MermaidStateDiagram] that initialized the diagram of
+  /// mermaid as `stateDiagram`.
+  ///
+  /// Passed output path to save in custom directory.
+  MermaidStateDiagram({this.outputPath = 'stateDiagram.md'})
+      : _outputFile = File(outputPath) {
+    diagram = 'stateDiagram\n';
+  }
+
+  /// Register a new state to the mermaid diagram object.
+  void addState(String state) => diagram = '$diagram\n$state\n';
+
+  /// Register a new transition [event] that point the
+  /// current state [currState] to next state [nextState].
+  void addTransitions(String currState, String nextState, String event) =>
+      diagram = '$diagram$currState --> $nextState: $event\n';
```

Comment:
I think we can use a `StringBuffer` instead of re-setting the `diagram` variable each time